analyzers/src/SonarAnalyzer.Common/Rules/Utilities/UtilityAnalyzerBase.cs

```diff
@@ -82,45 +83,62 @@ public abstract class UtilityAnalyzerBase<TSyntaxKind, TMessage> : UtilityAnalyz
 
         protected virtual bool AnalyzeUnchangedFiles => false;
 
-        protected virtual IEnumerable<TMessage> CreateAnalysisMessages(CompilationAnalysisContext c) => Enumerable.Empty<TMessage>();
+        protected virtual IEnumerable<TMessage> CreateAnalysisMessages(Compilation compilation) => Enumerable.Empty<TMessage>();
 
         protected UtilityAnalyzerBase(string diagnosticId, string title) : base(diagnosticId, title) { }
 
         protected sealed override void Initialize(SonarAnalysisContext context) =>
-            context.RegisterCompilationAction(c =>
+            context.RegisterCompilationStartAction(startContext =>
+            {
+                ReadParameters(context, startContext.Options, startContext.Compilation);
+                if (!IsAnalyzerEnabled)
```

Comment:
Can we get rid of this property and return `bool` from `ReadParameters` instead?

Reply:
This causes trouble with testability. See https://github.com/SonarSource/sonar-dotnet/pull/6576#discussion_r1060049807

Reply:
Why for this comment? The `ReadParameters` method would still be preserved. Only the usage would change

Reply:
Many of my old comments don't apply anymore because of your refactoring. This is one of the comments. I still would not return `bool` from `ReadParameters` to indicate "IsAnalyzerEnabled". The ReadParameters methods sets a lot of properties and IsAnalyzerEnabled is just one of them. Also the tests work like this:

* Set "IsAnalyzerEnabled = true" in the constructor
* In the "Initialize" method "ReadParameters" is called
* "ReadParameters" does only IsAnalyzerEnabled to true, but never to false and so the test setting is always preserved.

I don't see how this could be changed easily. If we want to change it, we should do so in another PR.